### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DmytroBondariev**

### CODE REVIEW

File: cinema/views.py [hunk 1]
```diff
@@ -101,8 +104,22 @@ def get_serializer_class(self):
...
+
         return MovieSerializer
 
+    @action(methods=["POST"], detail=True, url_path="upload-image")
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+
+        if serializer.is_valid():
+            serializer.save()
```
Comment (vsmutok) [inline cinema/views.py]: use serializer.is_valid(raise_exception=True)

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -7,4 +7,4 @@ venv/
 .pytest_cache/
 **__pycache__/
 *.pyc
-**db.sqlite3
+**db.sqlite3
```
Comment (ArsenPidhoretskyi) [inline .gitignore]: Restore this changes

File: cinema/models.py [hunk 3]
```diff
@@ -35,12 +39,24 @@ def full_name(self):
         return f"{self.first_name} {self.last_name}"
 
 
+def movie_image_file_path(instance, filename):
+    _, extension = os.path.splitext(filename)
+    filename = f"{slugify(instance.title)}-{uuid.uuid4()}.{extension}"
```
Comment (ArsenPidhoretskyi) [inline cinema/models.py]: Is dot really needed? Did you test manually?
```suggestion
    filename = f"{slugify(instance.title)}-{uuid.uuid4()}{extension}"
```

File: cinema/models.py [hunk 4]
```diff
@@ -35,12 +39,24 @@ def full_name(self):
...
+
+    return os.path.join("uploads/movies", filename)
```
Comment (ArsenPidhoretskyi) [inline cinema/models.py]:
```suggestion
    return os.path.join("uploads", "movies", filename)
```
Better to do like this. It will works in all OS

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**